### PR TITLE
Feature/add verify

### PR DESCRIPTION
**Pull request by joelamouche**
solves https://github.com/polkadot-js/apps/issues/4005

Thanks to https://github.com/polkadot-js/common/pull/767 ethereum signature verifications work.

I added the right verification in the verify component (ecdsa). This also works for non-ethereum ecdsa signatures

### CODE REVIEW

**Comment by jacogr** [comment]
Sorry for the absolutely long delay, not on purpose - I just merged the update with the common libs so the verify now (as fixed by you), should be operational... hopefully :)

**Comment by joelamouche** [comment]
No problem, thank you!

**Comment by joelamouche** [comment]
@jacogr ready for review

File: packages/page-signing/src/Verify.tsx [hunk 1]
```diff
@@ -56,13 +59,26 @@ function Verify ({ className = '' }: Props): React.ReactElement {
         if (isValidSr) {
           cryptoType = 'sr25519';
         } else {
-          isValid = false;
+          let verification;
```
Comment (jacogr) [inline packages/page-signing/src/Verify.tsx:62]: Should probably just use `signatureVerify` here and get rid of the specifics for sr/ed. We get the valid & type for free.

(Should actually have been done before, as it stands it is a duplication of what we have in util-crypto)

File: packages/react-params/src/initValue.ts [hunk 2]
```diff
@@ -30,9 +30,7 @@ export default function getInitValue (registry: Registry, def: TypeDef): unknown
       : {};
   }
 
-  const type = [TypeDefInfo.Compact, TypeDefInfo.Option].includes(def.info)
-    ? (def.sub as TypeDef).type
-    : def.type;
+  const type = [TypeDefInfo.Compact, TypeDefInfo.Option].includes(def.info) ? (def.sub as TypeDef).type : def.type;
```
Comment (jacogr) [inline packages/react-params/src/initValue.ts:33]:
```suggestion
  const type = [TypeDefInfo.Compact, TypeDefInfo.Option].includes(def.info)
    ? (def.sub as TypeDef).type
    : def.type;
```

Assuming this snuck in via prettier?

File: packages/react-api/src/types.ts [hunk 3]
```diff
@@ -54,7 +54,9 @@ export interface CallState {
 
 export type CallProps = ApiProps & CallState;
 
-export type BaseProps<T> = BareProps & CallProps & ChangeProps & {
+export type BaseProps<T> = BareProps &
+CallProps &
+ChangeProps & {
```
Comment (jacogr) [inline packages/react-api/src/types.ts:59]:
```suggestion
export type BaseProps<T> = BareProps & CallProps & ChangeProps & {
```

Actually weird, thought we manager to get rid of all the type with `&` and replaced with `interface`. (Probably some old reason behind this one)

File: packages/react-params/src/initValue.ts [hunk 4]
```diff
@@ -140,7 +138,9 @@ export default function getInitValue (registry: Registry, def: TypeDef): unknown
       if (!warnList.includes(type)) {
         warnList.push(type);
         error && console.error(`params: initValue: ${error}`);
-        console.info(`params: initValue: No default value for type ${type} from ${JSON.stringify(def)}, using defaults`);
+        console.info(
+          `params: initValue: No default value for type ${type} from ${JSON.stringify(def)}, using defaults`
+        );
```
Comment (jacogr) [inline packages/react-params/src/initValue.ts:143]:
```suggestion
        console.info(`params: initValue: No default value for type ${type} from ${JSON.stringify(def)}, using defaults`);
```

Prettier? Very weird one.

**Review by jacogr — COMMENTED** [review]
Generally looks good. Think we should just simplify the verification with `signatureVerify` instead of re-doing that here. (Was duplication anyway before this, now it is just more)

File: packages/react-params/src/initValue.ts (hunk 2 above)
Comment (joelamouche) [reply packages/react-params/src/initValue.ts:33]: yes indeed. let me reverse that

File: packages/page-signing/src/Verify.tsx [hunk 5]
```diff
@@ -31,38 +33,18 @@ function Verify ({ className = '' }: Props): React.ReactElement {
...
-        }
+      if (verification.crypto !== 'none') {
+        isValid = verification.isValid;
+        cryptoType = verification.crypto;
       }
     }
 
     setValidity({ cryptoType, isValid });
-  }, [data, isValidPk, isValidSignature, publicKey, signature]);
+  }, [isEthereum, data, isValidPk, isValidSignature, publicKey, signature]);
```
Comment (jacogr) [inline packages/page-signing/src/Verify.tsx:47]:
```suggestion
  }, [data, isValidPk, isValidSignature, publicKey, signature]);
```

Think this is not used now?

**Review by jacogr — APPROVED** [review]

**Comment by jacogr** [comment]
As the custom by now, thanks a million for this.

**Pull request merged**